spyder/api/plugins/new_api.py

```diff
@@ -1092,327 +1664,557 @@ def __init__(self, parent, configuration):
+        ----------
+        force_focus : bool | None, optional
+            If ``True``, always give the plugin's widget focus when showing
+            or hiding it with this method. If ``None``, only give it focus when
+            showing, not hiding (i.e. setting ``state`` to ``True``),
+            and only if :attr:`RAISE_AND_FOCUS` is ``True``. If ``False``,
+            the default, don't give it focus regardless.
+
+        Returns
+        -------
+        None
         """
         if self.get_widget().windowwidget is None:
             self.sig_switch_to_plugin_requested.emit(self, force_focus)
 
-    def set_ancestor(self, ancestor_widget):
+    def set_ancestor(self, ancestor_widget: QWidget) -> None:
         """
         Update the ancestor/parent of child widgets when undocking.
+
+        Parameters
+        ----------
+        ancestor_widget: QWidget
+            The window widget to set as a parent of this one.
+
+        Returns
+        -------
+        None
         """
         self.get_widget().set_ancestor(ancestor_widget)
 
     # ---- Convenience methods from the widget exposed on the plugin
     # -------------------------------------------------------------------------
     @property
-    def dockwidget(self):
+    def dockwidget(self) -> spyder.widgets.dock.SpyderDockWidget:
+        """The dockable widget (pane) for this plugin."""
         return self.get_widget().dockwidget
 
     @property
-    def options_menu(self):
+    def options_menu(self) -> PluginMainWidgetOptionsMenu:
+        """The options ("hamburger") menu widget for this plugin."""
         return self.get_widget().get_options_menu()
 
     @property
-    def toggle_view_action(self):
+    def toggle_view_action(self) -> spyder.utils.qthelpers.SpyderAction:
+        """The :class:`QAction` for showing/hiding this plugin's pane."""
         return self.get_widget().toggle_view_action
 
-    def create_dockwidget(self, mainwindow):
+    def create_dockwidget(
+        self,
+        mainwindow: spyder.app.mainwindow.MainWindow,
+    ) -> spyder.widgets.dock.SpyderDockWidget:
+        """
+        Create a new dock widget for this plugin.
+
+        Parameters
+        ----------
+        mainwindow : spyder.app.mainwindow.MainWindow
+            The main window to set as the dockwidget's parent.
+
+        Returns
+        -------
+        spyder.widgets.dock.SpyderDockWidget
+            The new dock widget created for the plugin.
+        """
         return self.get_widget().create_dockwidget(mainwindow)
 
-    def create_window(self):
+    def create_window(self) -> None:
+        """
+        Create a window for the plugin.
+
+        Returns
+        -------
+        None
+        """
         self.get_widget().create_window()
 
-    def close_window(self, save_undocked=False):
+    def close_window(self, save_undocked: bool = False) -> None:
+        """
+        Close the plugin's window, optionally saving the undocked state.
```

Comment:
```suggestion
        Close the plugin's undocked window, optionally saving its state.
```

To go with my suggestion for `create_window`.

Reply:
Only potential concern here is being specific about _what_ state it is saving (whether it is undocked/docked, its undocked position, size, etc., other state aspects?)